homeassistant/components/esphome/assist_satellite.py

```diff
@@ -180,11 +192,57 @@ async def async_set_configuration(
         # Ensure configuration is updated
         await self._update_satellite_config()
 
+    def _get_custom_wake_words(self) -> dict[str, VoiceAssistantExternalWakeWord]:
+        """Get a list of custom wake words."""
+        wake_words = {}
+        for config_path in self._wake_words_dir.glob("*.json"):
+            wake_word_id = config_path.stem
+
+            with open(config_path, encoding="utf-8") as config_file:
+                config_dict = json.load(config_file)
+                if not (model := config_dict.get("model")) or (
+                    not (model_path := (self._wake_words_dir / model)).exists()
+                ):
+                    # Missing model file
+                    continue
+
+                if not (model_type := config_dict.get("type")) or (
+                    not (wake_word := config_dict.get("wake_word"))
+                ):
+                    # Invalid config
+                    continue
+
+                with open(model_path, "rb") as model_file:
+                    model_hash = hashlib.sha256(model_file.read()).hexdigest()
+
+                model_size = model_path.stat().st_size
+                config_rel_path = config_path.relative_to(self._wake_words_dir)
+                base_url = get_url(self.hass)
+
+                wake_words[wake_word_id] = VoiceAssistantExternalWakeWord.from_dict(
+                    {
+                        "id": wake_word_id,
+                        "wake_word": wake_word,
+                        "trained_languages": config_dict.get("trained_languages", []),
+                        "model_type": model_type,
+                        "model_size": model_size,
+                        "model_hash": model_hash,
+                        "url": f"{base_url}/api/esphome/wake_words/{config_rel_path}",
+                    }
+                )
+
+        return wake_words
+
     async def _update_satellite_config(self) -> None:
         """Get the latest satellite configuration from the device."""
+        self._external_wake_words = await self.hass.async_add_executor_job(
+            self._get_custom_wake_words
```

Comment:
it's weird that we scan the directory for every wake word. Can't we use some cache, especially during startup ?